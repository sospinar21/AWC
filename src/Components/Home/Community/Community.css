.post-container {
  width: 100%;
  height: 92%;
  overflow-y: scroll; }

.post {
  width: 90%;
  background-color: #f9f9f9;
  margin: auto;
  height: 15vh;
  margin: 2vh auto;
  box-shadow: 0px 0px 2px #e8e2e282;
  padding: 1.5vh;
  box-sizing: border-box;
  display: flex; }

.post-img {
  width: 90%;
  height: 100%;
  background-image: url("../../../assets/event.png");
  background-size: cover; }
  .post-img img {
    width: 100%;
    height: 100%; }

.text-description {
  margin: 0 2vh;
  font-size: 1.8vh;
  height: 70%;
  overflow: scroll; }
  .text-description p {
    letter-spacing: .35px;
    line-height: 1.2; }

.likes {
  text-align: right; }

.icon {
  line-height: 0;
  margin: 2vh; }
