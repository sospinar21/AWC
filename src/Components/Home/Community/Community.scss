.post-container {
  width: 100%;
  height: 92%;
  overflow-y: scroll;
}

.post {
  width: 90%;
  background-color: #f9f9f9;
  margin: auto;
  height: 15vh;
  margin: 2vh auto;
  box-shadow: 0px 0px 2px #e8e2e282;
  padding: 1.5vh;
  box-sizing: border-box;
  display: flex
}

.post-img{
  width: 20%;
  height: 100%;
  background-image: url('../../../assets/event.png');
  background-size: cover;

  img {
    width: 100%;
    height: 100%;
  }
}

.post-description {

  h3 {
    margin: 1vh;
    color: #b62d4c;
  }
}

.text-description {
  margin: 0 2vh;
  font-size: 1.8vh;
  height: 40%;
  overflow: scroll;

  p {
    letter-spacing: .35px;
    line-height: 1.2;
  }
}

.likes {
  text-align: right;
}

.icon {
  line-height: 0;
  margin: 2vh;

}