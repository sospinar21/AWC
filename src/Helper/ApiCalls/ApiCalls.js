import React, {Component} from 'react';

class ApiCalls extends Component {
  constructor(){
    super()
  }

  fetchEvents = async () => {
    const url = `https://api.awc.dance/events?city=vermont`;
    const response = await fetch(url)
    const data= await response.json();
    return data;
  } 

  fetchStudios = () => {
    const url = `https://api.awc.dance/?city=80202`;
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-type': 'application/json',
      }
    }).then( response => response.json() )
      .then( data => data )   
  }

  fetchVideos= async () => {
    var url = 'https://api.awc.dance/youtubevideos'
    const response = await fetch(url)
    const data = await response.json()
    return data
  }
}

export default ApiCalls;