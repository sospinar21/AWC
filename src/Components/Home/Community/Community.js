import React, { Component } from 'react';
import './Community.css';
import  Post  from '../Post/Post';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


export class Community extends Component {
  constructor() {
    super(); 
    this.state = {
      posts: []
    };
  }

  addPost = (post) => {
    const posts = [post, ...this.state.posts];
    this.setState({posts});
  }
  
  displayForum = () => {
    const user = this.props.user.username;
    if (this.state.posts.length){
      const posts = this.state.posts.map((post, index) => {
        return (
          <div className='post' key={post.user + index}>
            <div className='post-img'>
            </div>
            <div className='post-description'>
              <div className='text-description'> 
                <p>{post.input}</p>
              </div>
              <div className='likes'>
                <h3>category: {post.category}</h3>
                <h3>@ {user}</h3>
                {/* <p>{post.likes}</p><i className="material-icons icon" onClick={(e)=> this.upvote(e, post)}>thumb_up_alt</i>
                <p>{post.dislikes}</p><i className="material-icons icon" onClick={(e)=> this.downVote(e, post)}>thumb_down_alt</i> */}
              </div>
            </div>
          </div>
        );
      });
      return posts;
    }
  }

  render () {
    return (
      <div className='post-container'>
        <Post addPost = {this.addPost}/>
        {this.displayForum()}
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return ({
    user: state.user
  });
};

export const mapDispatchToProps = dispatch => ({
});

Community.propTypes = {
  user: PropTypes.obj
};

export default connect(mapStateToProps, mapDispatchToProps)(Community);