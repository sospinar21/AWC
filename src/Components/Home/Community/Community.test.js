import {Community, mapStateToProps} from '../Community/Community';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';

describe('community', () => {
  let community;
  let mockProps;
  const api = new ApiCalls();

  beforeEach(() => {
    mockProps = {
      user: {username: 'Steph'},
    };

    api.getPosts = jest.fn();
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({data: {}})
    }));

    community = shallow(<Community {...mockProps}/>, { disableLifecycleMethods: true });

  });

  it('matches snapshot', () => {
    let community = shallow(<Community {...mockProps}/>, { disableLifecycleMethods: true });
    expect(community).toMatchSnapshot();
  });

  describe('componenDidMount', () => {

    it('calls api get post on load', () => {
      let community = shallow(<Community {...mockProps}/>);

      api.getPosts.mockImplementation(() => Promise.resolve({
        status: 200,
        json: () => Promise.resolve({data: []})
      }));

      community.instance().componentDidMount();
      expect(community.state().posts).toEqual([])
    });
  });

  describe('addPost', () => {
  
    it('sets the state with the list of posts', () => {
      community.state().posts = []
      const post = {content:'hello'}
      const expected = [{content:'hello'}]

      community.instance().addPost(post);

      expect(community.state().posts).toEqual(expected);
    });
  });

  describe('mapStateToProps', () => {

    let mappedProps

    it('returns an object with the user', () => {

      const mockState = {
        user: {username: 'Kai'}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })

  

});