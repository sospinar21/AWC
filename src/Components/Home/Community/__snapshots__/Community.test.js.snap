// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`community matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Community
    user={
        Object {
            "username": "Steph",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Post)
          addPost={[Function]}
/>,
        <div
          className="post-container"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addPost": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "post-container",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Post)
            addPost={[Function]}
/>,
          <div
            className="post-container"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addPost": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "post-container",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
